src/amr/data/particles/refine/split_2d.h

```diff
@@ -0,0 +1,172 @@
+#ifndef PHARE_SPLIT_2D_H
+#define PHARE_SPLIT_2D_H
+namespace PHARE::amr
+{
+template<> // Generic 2D delta setter
+struct SplitInnerSetter</*dim=*/2, 0>
+{
+    template<typename Deltas, typename DeltaVal>
+    static void set_deltas(Deltas& deltas, DeltaVal& delta_vals, size_t root)
+    {
+        for (size_t y = 0; y < root; y++)
+            for (size_t yoff = y * root, x = 0; x < root; x++)
+                deltas[x + yoff] = {delta_vals[x], delta_vals[y]};
+    }
+};
+
+
+/*************************************************************************
+  dim = 2
+  interp = 1
+  nbrRefineParticles = 9
+*/
+template<>
+class Splitter<2, 1, 9> : public ASplitter<2, 1, 9>
+{
+public:
+    Splitter()
+        : ASplitter{weight_vals_, delta_vals_}
+    {
+    }
+
+protected:
+    constexpr static float delta_vals_[]  = {-1, 0, 1};
+    constexpr static float weight_vals_[] = {0.25, 0.125, 0.01625};
+};
+
+template<>
+struct SplitInnerSetter</*dim=*/2, /*nbrRefineParticles*/ 9>
+{
+    constexpr static size_t dim  = 2;
+    constexpr static size_t root = 3; // 3*3
+
+    template<typename Weights, typename WeightsVal>
+    static void set_weights(Weights& weights, WeightsVal& weight_vals)
+    {
+        weights[4] = weight_vals[0];
+        for (size_t i : {0, 2, 6, 8})
+            weights[i] = weight_vals[2];
+        for (size_t i : {1, 3, 5, 7})
+            weights[i] = weight_vals[1];
+    }
+
+    template<typename Deltas, typename DeltaVal>
+    static void set_deltas(Deltas& deltas, DeltaVal& delta_vals)
+    {
+        SplitInnerSetter<dim, 0>::set_deltas(deltas, delta_vals, root);
```

Comment:
Since I don't see this method working for the approached splitting (where not all particles are dispatched)
I think it's more reasonable to set deltas per pattern.

let's take for instance dim2 order 2

I think there are 3 patterns:

green : 
delta=(0.5, 0.5), weight 0.140625

orange:
delta= (0.5, 1.5) and (1.5,0.5) and weight 0.046875

red:
delta = (1.5,1.5), weight = 0.015625

actually here I don't remember how @rochSmets counts, maybe he says there are two patterns because green and red are the same just with different deltas and weights.

Reply:
so that means we consider there are 3 patterns here ?

Reply:
yes... 3 patterns with 3 weights : 0.375*0.375, 0.375*0.125 and 0.125*0.125.
for the delta's : (0.5, 0.5), (0.5, 1.5) & (1.5 , 0.5) (1.5, 1.5).